### PR TITLE
add patch for not linking to clang rt bulitin lib, and apply patches to 3.9.0 too

### PR DESCRIPTION
**Pull request by haampie**
"Backports" the git-version-info-patch to 3.9.0 and 3.10.0, since they are still used a lot.

Also adds a patch for not linking against clang runtime libs in hip::host exported target, which was considered a mistake: https://github.com/ROCm-Developer-Tools/HIP/pull/2219

### CODE REVIEW

**Comment by haampie** [comment]
@srekolam / @arjun-raj-kuppala are you OK with these changes?

- hip::host does not link to clang bulitins lib (approved here: https://github.com/ROCm-Developer-Tools/HIP/pull/2219)
- fixes the lib sonames for hip, because the version was previously derived from git, and we dont have the repo
- expose LLVM_PATH as a runtime env variable, as it is used in comgr at runtime
- move patches into the `def patch():` function to avoid re-applying those patches when you dev-build hip in spack.

File: var/spack/repos/builtin/packages/hip/0003-Improve-compilation-without-git-repo.3.9.0.patch [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
++++ b/CMakeLists.txt
+@@ -44,9 +44,17 @@ string(REPLACE "." ";" VERSION_LIST ${HIP_BASE_VERSION})
+ list(GET VERSION_LIST 0 HIP_VERSION_MAJOR)
+ list(GET VERSION_LIST 1 HIP_VERSION_MINOR)
+ 
+-find_package(Git)
++# only look for git when we have a git repo
++if (IS_DIRECTORY "${PROJECT_SOURCE_DIR}/.git")
++  find_package(Git)
++endif()
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/hip/0003-Improve-compilation-without-git-repo.3.9.0.patch:22]: :+1:

**Review by alalazo — APPROVED** [review]
LGTM. @srekolam @arjun-raj-kuppala Can you have a look and confirm it's fine with you too?

File: var/spack/repos/builtin/packages/hip/0003-Improve-compilation-without-git-repo.3.9.0.patch (hunk 1 above)
Comment (srekolam) [reply var/spack/repos/builtin/packages/hip/0003-Improve-compilation-without-git-repo.3.9.0.patch:22]: i have looked at this changes but thought i will get this reviewed with some one more knowledgeable on the hip..allow me couple of days to revert back on this if possible

**Comment by srekolam** [comment]
Reviewed with the internal hip team and Okaying it

**Review by srekolam — APPROVED** [review]

**Pull request merged**